src/item.cpp

```diff
@@ -3719,16 +3722,36 @@ std::set<matec_id> item::get_techniques() const
 
 bool item::goes_bad() const
 {
+    if( is_corpse() ) {
+        // Corpses rot only if they are made of rotting materials
+        std::vector<std::string> rotting_materials = {"Insect Flesh", "Human Flesh", "Flesh", "Bone"};
+        for( std::string material : rotting_materials ) {
+            if( made_of_types()[0]->name() == material ) {
```

Comment:
`->name()` returns translated string while `rotting_materials` is untranslated. I suspect it won't work properly when language is not set to English and material names are translated in that language and performance is very bad.

`made_of_types` returns `std::vector<const material_type *>`, so we better make `rotting_materials` of  same type and don't compare strings.